### PR TITLE
Goodbye, IE 11

### PR DESCRIPTION
**Pull request by noahtallen**
### Changes proposed in this Pull Request

Enable redirect for fallback browsers in all environments. Follow-up from #53221, which implemented the redirect for development and some testing environments. I also took the opportunity to remove documentation referencing IE11 and the fallback build.

See #53222 for follow-up tasks.

### Testing instructions

Use calypso.live for the following:

- Verify that IE 11 is redirected to `/browsehappy`. Browserstack works well for a quick test. (Just load the calypso.live link in IE 11. You can also test the "continue anyways" path and ensure it does not redirect after that.
- Verify that supported browsers are not redirected. (Load calypso.live in something like chrome or firefox and ensure that you do not get redirected.

We will have to double check this behavior after shipping once the deployment makes it to staging, and then again after it gets to prod, as this code change modifies environment configuration.

### CODE REVIEW

File: client/server/middleware/unsupported-browser.js [hunk 1]
```diff
@@ -1,15 +1,9 @@
 /**
  * Internal dependencies
  */
-import config from '@automattic/calypso-config';
 import { addQueryArgs } from 'calypso/lib/url';
 
 export default () => ( req, res, next ) => {
-	if ( ! config.isEnabled( 'redirect-fallback-browsers' ) ) {
```
Comment (scinos) [inline client/server/middleware/unsupported-browser.js]: I'd suggest we set 'redirect-fallback-browsers:true` in prod environments first, and after having it for a few days without incidents, then delete the flags.

Comment (noahtallen) [reply client/server/middleware/unsupported-browser.js]: Done!

**Comment by worldomonation** [comment]
Reminder to myself to scan and remove any parts about IE11 in the `test/e2e` side of things.

**Review by lynnbgriffith — APPROVED** [review]
👍 Tests well for me in IE11 and a number of modern browsers!

**Pull request merged**